### ISSUE TITLE
[Android] Import passwords from chrome

### ISSUE THREAD

**Issue by deeppandya** [brave/brave-browser#35729]
This issue is to import password from chrome. 
In chrome we have an option to export passwords in .csv format as shown below, 
![Screenshot_20240131-114029](https://github.com/brave/brave-browser/assets/5146988/630fc983-fa70-4fbf-9771-98ba9d623bbb)

In Brave, we can use .csv to import password into the app.

**Comment by nudeln** [comment]
How can there be an export password feature, but no import? Should be as easy peacy to fix?

**Comment by Brave-Matt** [comment]
All sorts of +1s for this issue (this was just a quick search, can easily find many many more requests for this):
https://community.brave.com/t/how-to-import-passwords-from-chrome/343881/21
https://community.brave.com/t/how-to-import-passwords-from-chrome-etc/105015
https://community.brave.com/t/how-to-migrate-my-bookmarks-and-passwords-from-chrome-to-brave-browser/291425
https://community.brave.com/t/import-passwords-from-brave-desktop-to-mobile-android/234137
https://community.brave.com/t/import-password-without-comp/560235
https://community.brave.com/t/import-passwords-into-brave/545815
https://community.brave.com/t/brave-password-import/542524
https://community.brave.com/t/how-to-import-my-passwords-on-brave-browser-on-my-martphone/497332
https://community.brave.com/t/import-passwords-to-browser-brave-in-android/466732
https://community.brave.com/t/import-passwords-on-android/450306
https://community.brave.com/t/how-to-export-copy-passwords-from-android-to-mac-without-sync/283254/3

This is an extremely common feature request that I've been seeing since I was first hired here. Not unlike the old "[import bookmarks](https://github.com/brave/brave-browser/issues/11485)" issue, this is a big blocker/barrier for users wanting to move over from Chrome. 

Is there any way we can get this prioritized or at least get an update on the status here? cc @timchilds @anthonypkeane

**Comment by gabrielparca** [comment]
This really is a deeply needed feature... not only, but also while I have no way of toggling the "passwords" option on Sync...

![Brave not syncing psswords com-video-to-gif-converter](https://github.com/user-attachments/assets/6fdc6a49-8e74-4022-8581-f137d5c210c7)

**Comment by aguscruiz** [comment]
Adding Figma file for design reference:
https://www.figma.com/design/vHUet2idQQeyhANlnLaaOJ/Import-passwords?node-id=1-2

**Comment by JDF201** [comment]
How long til this beta shit will be fixed?

**Comment by AltanDaniel** [comment]
I made an account just to add my voice to this support for such a simple feature that has been asked for for years and denied. How can brave be so great in so many ways but be missing this important feature which enables us to migrate from one device to another?!

**Comment by AltanDaniel** [comment]
And yes, I jumped through all the hoops to enable sync on my old Samsung device, my new OnePlus device, AND the laptop computer that I for some reason need to use... and nothing. It still didn't sync my passwords, bookmarks, anything. This is ridiculous!

**Comment by AltanDaniel** [comment]
I finally managed to jump through all the hoops to add both my mobile devices - the old samsung, the new oneplus. And for some reason, of course the QR code scan did nothing - didn't give me a button to click, didn't give me a url, nothing. Then I had to type or dictate a long cumbersome list of words very carefully, still nothing. 

Then looking in the comments I realized for some reason I need to add a laptop or other large computer to the chain, I have no idea why. Finally after adding that computer, things synced after waiting for about a minute. 

But I should not have to jump through all those hoops to do such a simple task. If you claim it's for security, that's ridiculous because simply exporting all the passwords into a CSV file breaks any kind of security you were claiming to have in the first place. Why not simply allow the option to import those passwords that are already in an insecure text file and then offer to securely delete the file for them? How can you ignore users when they ask for something this simple for years?

**Comment by WalkEmDowm** [comment]
My first half baked idea was that if you could theoretically downloaded an early enough version of brave, you could take advantage of the option to import passwords, after which updating to the latest version. Unless I hallucinated having that feature in the past (which is just as likely...👻)
If anyone attempts this or actually knows what their talking about please enlighten me because this would be the official method for Android-only password imports.

For the time being the most accurately functioning, interim solution that I can attest to gets installed & functions like this: 
1. First download Brave Nightly
2. Next follow this link: [brave://flags/#enable-autofill-virtual-view-structure](url)
3. That takes you to [brave://flags](url) and the setting highlighted is the one you need to change to **Enable without AwG restriction**
4. Now you should at least have your passwords autofill directly from your keyboard. You might need Gboard installed, or at least just have autofill set up in your phone settings. 

On a side-note why is this feature disabled by default?
I don't understand why Android users are required to jump through all these hoops for a basic function that should be a standard feature, especially when you have new users that are trying to cut ties with chrome.
I'd be interested to know if this problem stems from Braves' side or Google.. so if you know.. enlighten me.

& For N-E-one that has tried either of these workarounds let me know how they work for 🫵

**Comment by bsclifton** [comment]
cc: @timchilds for prioritization

**Comment by gabrielparca** [comment]
PSA: this is also being discussed at the [Brave Comunity](https://community.brave.com/t/cant-turn-on-password-sync/559348/17?u=gparca) forum.

**Comment by Olipro** [comment]
I have a basic PoC working which imports and then displays a simple dialog to tell you how many entries were imported.

As a quick heads-up (without wanting to be _too_ premature) - I'm not entirely convinced the Chromium code populates the list of successfully imported entries. This is of note because the Figma mock-up specifically calls for a results screen that shows you what was imported and what wasn't.

Consequently:

* If it does populate them, additional code would be needed to also produce a list of what wasn't imported.
* If it doesn't populate them, additional code would be needed to make it do that and do 1. as well.

With that in mind, I foresee two approaches:

1. Forego the additional dialog that reports on import results, no need to start making any patches to Chromium's password porter code in C++ and this feature remains pretty uninvasive.
2. Implement the additional dialog and accept the additional work necessary to make C++-land do what we require.

If we choose option 2, I'd appreciate a discussion on how we'd like to _arrange the furniture_ with regard to achieving it - specifically with regard to the fact we already have the Safari importer which copies a fair amount of Chromium code which I'm reluctant to take another copy of for Android when it could be refactored into something more minimal in nature and adhere to the paradigm of DRY.

As a disclaimer: Perhaps I've completely missed seeing some code somewhere that makes getting what we need from C++-land absolutely trivial. If that's the case, please point me to it.

TL;DR: Option 1 is practically free and can be submitted for review in a day or two. Option 2 is not, even if we take a _quick-and-dirty_ approach - which I would be loathe to entertain (though doing it cleanly and properly is fine by me of course.)

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 16 BP41.250822.010 release-keys
Brave build: 1.84.124
Chromium: 141.0.7390.108 (Official Build) beta (64-bit)
```

Filed follow-up issue #50232, #50245
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-core/pull/31056#issue-3387739823
2. *Verify*

**ACTUAL RESULTS:**
- Verified that passwords can be exported.
- Verified that passwords can be imported:
  - Verified that passwords exported from `Brave` can be imported in `Brave`.
  - Verified that passwords exported from `Chrome` can be imported in `Brave`.
- Verified that the lists of saved passwords and exceptions continue to function as they did before.

---------------------------

| 1 | 2 | 3 | 4 |
|:-:|:-:|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/e5f44385-404e-445d-b598-e9a995738282) | ![2](https://github.com/user-attachments/assets/ca0799f9-0c79-4a29-b0de-3d025f187971) | ![3](https://github.com/user-attachments/assets/9531b622-3059-4e73-819b-b19b86899f89) | ![4](https://github.com/user-attachments/assets/2b5d0b07-d8fb-48b6-85cb-96c3047e62eb) |
| ![1](https://github.com/user-attachments/assets/5c64627b-09d2-4afa-a5ef-27bcbc3994b3) | ![2](https://github.com/user-attachments/assets/0922a1af-2e7e-4de0-aa40-5bd1da1be04a) | ![3](https://github.com/user-attachments/assets/b9f56ba1-dc7e-4747-a55f-e3c9cb17ab52) | ![4](https://github.com/user-attachments/assets/2ebe84e5-cb7d-43a7-bf6a-12befc5eb8fd) |
| ![1](https://github.com/user-attachments/assets/14b81af9-065a-4a96-b992-1c922d500528) | ![2](https://github.com/user-attachments/assets/11838b4a-b04c-4425-a3c1-71318e6e20ea) | ![3](https://github.com/user-attachments/assets/31c87738-2194-4ab1-af89-91ebf93d6d6e) | ![4](https://github.com/user-attachments/assets/c36a3e1f-1a62-430a-b1fd-c9ce2151d56a) |
| ![1](https://github.com/user-attachments/assets/0ce5b9f8-bb9a-4069-bc1a-44d4a84d3a38) | ![2](https://github.com/user-attachments/assets/26cacf10-5105-42ee-b418-f3bbe7f3c4d1) | ![3](https://github.com/user-attachments/assets/bc419c88-f3c2-4832-ac59-5b576f168824) | ![4](https://github.com/user-attachments/assets/f3e3252b-1b70-4da0-aec8-9d866a48c58c) |
| ![1](https://github.com/user-attachments/assets/585b8793-d017-41c8-804c-368f8630e47b) | ![2](https://github.com/user-attachments/assets/b1e0e66f-da8c-4a3d-8b1b-4e1e4b31f06f) | ![3](https://github.com/user-attachments/assets/dc9a3465-7efe-4330-8dfa-53bb4eb9bcf1) | ![4](https://github.com/user-attachments/assets/d65bf0db-f38e-4f3a-85e0-7acf51fdf9dd) |

**Comment by hffvld** [comment]
Verified on `Galaxy Tab S8` using version(s):
```
Device/OS: Galaxy Tab S8 / gts8wifixx-user 15 AP3A.240905.015.A2 release-keys
Brave build: 1.84.127
Chromium: 141.0.7390.108 (Official Build) beta (64-bit)
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-core/pull/31056#issue-3387739823
2. *Verify*

**ACTUAL RESULTS:**
- Verified that passwords can be exported.
- Verified that passwords can be imported:
  - Verified that passwords exported from `Brave` can be imported in `Brave`.
  - Verified that passwords exported from `Chrome` can be imported in `Brave`.
- Verified that the lists of saved passwords and exceptions continue to function as they did before.

---------------------------

| 1 | 2 | 3 | 4 |
|:-:|:-:|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/7b8e2d91-6391-48a8-80c5-b7a4efd22a19) | ![2](https://github.com/user-attachments/assets/65c80885-79d6-4cf0-9eb4-f7bed8e0a200) | ![3](https://github.com/user-attachments/assets/e99c4edb-f84a-4eb6-9995-29ab55ecc0e2) | ![4](https://github.com/user-attachments/assets/ed9a55da-3ef3-45d2-a9bb-36fcc9bc8f17) |
| ![1](https://github.com/user-attachments/assets/701ec92e-482f-497c-acba-da8c2869b1e3) | ![2](https://github.com/user-attachments/assets/fc69bf7d-a669-45cb-ae91-e8fd928c0238) | ![3](https://github.com/user-attachments/assets/96adb448-a0bb-4f0f-94cf-13ccdd43ad2c) | ![4](https://github.com/user-attachments/assets/03ac2cdd-81b0-4057-b9c9-7d578b2bbe3e) |
| ![1](https://github.com/user-attachments/assets/33c9b613-2655-4b7c-852b-c7159e105975) | ![2](https://github.com/user-attachments/assets/da2fce46-1111-49ff-be2c-bb3dec3bf2c3) | ![3](https://github.com/user-attachments/assets/c27c4636-ab58-4f94-a1ef-59f11469485e) | ![4](https://github.com/user-attachments/assets/982ea904-98a8-4337-9b5d-5490e3828875) |
| ![1](https://github.com/user-attachments/assets/923757fb-1fcb-4b85-86bc-a5032e41df81) | ![2](https://github.com/user-attachments/assets/1ddf5bdc-27f3-4b2d-9894-ffc25c2e156d) | ![3](https://github.com/user-attachments/assets/df4462e1-c51a-4200-b709-1cab5a7be4d4) | ![4](https://github.com/user-attachments/assets/0ba0953f-4962-4af7-86cc-75d3ac897935) |
| ![1](https://github.com/user-attachments/assets/ff7a3eee-4b04-4ce6-ac72-f2448901c8a8) | ![2](https://github.com/user-attachments/assets/7b93e907-f72d-4620-ba5f-c080b7ca7b7a) | ![3](https://github.com/user-attachments/assets/dc257dfb-7c15-442a-9d9a-afa5caebacbd) | ![4](https://github.com/user-attachments/assets/9a5c3005-47fa-4ca0-9095-b1c69d9f76a2) |
| ![1](https://github.com/user-attachments/assets/fc758d0b-9ab5-4db7-86e7-0706758a6263) | ![2](https://github.com/user-attachments/assets/0137c919-5ccc-4af3-9f4a-96243e938de5) | ![3](https://github.com/user-attachments/assets/3b7a2afa-7190-4443-a90e-b7f504262f33) | ![4](https://github.com/user-attachments/assets/403afaee-d69c-4e3b-abe7-62a2a4fd49da) |
